pkg/manifests/nmstateconfig.go

```diff
@@ -1,23 +1,25 @@
 package manifests
 
 import (
+	"context"
 	"fmt"
-	"os"
 	"reflect"
 
 	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
 	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/staticnetworkconfig"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
-func getNMStateConfig() aiv1beta1.NMStateConfig {
+func getNMStateConfig() (aiv1beta1.NMStateConfig, error) {
 	var nmStateConfig aiv1beta1.NMStateConfig
 	if err := GetFileData("nmstateconfig.yaml", &nmStateConfig); err != nil {
-		fmt.Println(err.Error())
-		os.Exit(1)
+		// This file is optional if DHCP addressing is used
+		return nmStateConfig, err
```

Comment:
This is temporary, right? We won't support fully dhcp envs any time soon

Reply:
Yes, as we talked about, only temporary until https://github.com/openshift-agent-team/fleeting/pull/22 merges